src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTagger.cs

```diff
@@ -19,56 +24,118 @@ internal sealed class InlineParameterNameHintsTagger : ITagger<IntraTextAdornmen
     {
         private readonly ITagAggregator<InlineParameterNameHintDataTag> _tagAggregator;
         private readonly ITextBuffer _buffer;
+        private readonly ITextView _textView;
+
+        /// <summary>
+        /// _cache stores the parameter hint tags in a global location 
+        /// </summary>
+        private readonly List<ITagSpan<IntraTextAdornmentTag>> _cache;
+
+        /// <summary>
+        /// Stores the snapshot of the state at which things change on the screen 
+        /// </summary>
+        private ITextSnapshot? _cacheSnapshot;
 
-        public event EventHandler<SnapshotSpanEventArgs> TagsChanged;
+        private readonly IClassificationFormatMap _formatMap;
+        private TextFormattingRunProperties? _format;
```

Comment:
Consider adding a doc comment to say this is lazy initialized, and to access it you should use the Format property.